File: ios/Packages/AppFeatures/Sources/SearchHistoryFeature/SearchHistoryModels.swift
import Foundation

public struct SearchHistoryEntry: Decodable, Equatable, Identifiable, Sendable {
    public let id: String
    public let source: String
    public let queryText: String
    public let locale: String
    public let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, source, locale
        case queryText = "query_text"
        case createdAt = "created_at"
    }
}

struct ListSearchHistoryResponse: Decodable {
    let entries: [SearchHistoryEntry]
}

struct RecordSearchRequest: Encodable {
    let source: String
    let query_text: String
    let locale: String
}

File: android/feature/searchhistory/src/main/kotlin/com/fatwabot/feature/searchhistory/SearchHistoryModels.kt
package com.fatwabot.feature.searchhistory

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SearchHistoryEntry(
    val id: String,
    val source: String,
    @SerialName("query_text") val queryText: String,
    val locale: String,
    @SerialName("created_at") val createdAt: String,
)

@Serializable
internal data class ListSearchHistoryResponse(val entries: List<SearchHistoryEntry>)

@Serializable
internal data class RecordSearchRequest(
    val source: String,
    @SerialName("query_text") val queryText: String,
    val locale: String,
)

@Serializable
internal data class DeletedResponse(val deleted: Boolean)

@Serializable
internal data class ClearedResponse(val cleared: Boolean)
